jsonschema-core.xml

```diff
@@ -125,57 +125,51 @@
             </t>
         </section>
 
-        <section title="Overview">
+        <!-- JSON Schema accomplishes two objectives, which each get their own section. -->
+        <section title="Validation">
             <t>
-                This document proposes a new media type "application/schema+json" to identify a JSON
-                Schema for describing JSON data.
-                It also proposes a further optional media type, "application/schema-instance+json",
-                to provide additional integration features.
-                JSON Schemas are themselves JSON documents.
-                This, and related specifications, define keywords allowing authors to describe JSON
-                data in several ways.
+                A JSON Schema document describes a validator (also known as a "recognizer" or "acceptor") which classifies a provided JSON document as "accepted" or "rejected."
```

Comment:
The "accept"/"reject" terminology is new.  I see you use it later in the PR as well, but it's not used throughout the document.

Reply:
It's new to this spec, but it is used widely outside JSON Schema and may help new readers understand what is going on. I'm going to suggest we should use accept/reject more often (it greatly simplifies the phrasing of many sentences), but that'll be an issue for later.

Reply:
Can you remove that language from this PR and open an issue for that change, please?

I'm not opposed to it, but I think vernacular should be an agreed-upon change, not something that's just snuck in.

Reply:
Well my point is there's a certain segment who may see _our_ language as new, and "accepts" is the existing term they're familiar with. I think we should use a variety of language to introduce and define the concepts, and then we can use our choice of term for the rest of the document. Is there a problem with this line of thinking?

Reply:
I don't have a problem with introducing them, but this PR doesn't seem the place for it.  I'd like to get the opinions of the other maintainers.

Reply:
To me the right question is "what's the most important thing that someone first reading the spec should see"?

I don't think what's here previously was right (because I'd assert most people reading the spec don't know or care what a media type is -- maybe to a lesser extent if we were publishing an IETF draft but certainly if we aren't). But I don't see this as being a step forward personally.

Very superficially to start, I think calling the section "Overview" is a better choice than the change (as it directs someone "hey look over here if you want to read something short"). Having subheadings may still be useful, but yeah, strictly removing the title makes it harder to skim.

I'd probably also question the assertion that an "acceptor/rejector/recognizer" language is widely used everywhere but by us. Which circles are you referring to, and again, regardless of those circles, is that the first thing someone should encounter when trying to answer "what is JSON Schema"?

All in all I find the first few paragraphs here to be a step back, I'd personally be trying to deemphasize the notion of a media type but sticking quite close to what was originally here, i.e. something like:

>                 This document proposes a JSON-based format for schematizing and annotating JSON documents, known as JSON Schema.
                JSON Schemas are themselves JSON documents.
                This, and related specifications, define keywords allowing authors to describe JSON
                data in several ways.
                Two media types for schemas are also defined.
                etc.

More generally, I'd say I'm personally reasonably comfortable with the CS concepts you seem to be trying to introduce into the spec, but whilst I feel I sufficiently understand them, I don't see them as adding *understanding* to someone reading the spec. Yes some of these things have names, but [as Feynman said about explaining science](https://bigthink.com/personal-growth/how-to-use-the-feynman-technique-to-identify-pseudoscience/):

> Suppose a student would say, ‘I don’t think energy makes [a dog] move.’ Where does the discussion go from there? I finally figured out a way to test whether you have taught an idea or you have only taught a definition. Test it this way: You say, ‘Without using the new word which you have just learned, try to rephrase what you have just learned in your own language. Without using the word “energy,” tell me what you know now about the dog’s motion.’ You cannot. So you learned nothing about science. That may be all right. You may not want to learn something about science right away. You have to learn definitions. But for the very first lesson, is that not possibly destructive?

I am of course not accusing this of being pseudoscience. But I do think that a lot of the terms you seem keen on introducing, whilst *accurate*, don't necessarily add any understanding -- someone familiar with them already quite possibly recognizes some of these things without them being mentioned. And someone who isn't, doesn't gain anything from seeing the word "context-sensitive" in the first paragraph about JSON Schema, nor from trying to understand what you mean by saying schemas define sets of documents. It's not inaccurate clearly in both cases, but it's not helpful to say so, it doesn't add enough *meaning*. My 0.02.

I suppose I'm leaving this comment on the first bit of the PR even though it applies to a bunch of what's below as well -- to me, what's here in this whole PR does too much, and does it without being clear on specific motivation for a number of the changes. Moving some discussion of vocabularies to sit before the introduction of concrete vocabularies I might agree with. Rewording the intro paragraphs to make them *more* of an overview I probably would to, in a less drastic way. But on introducing some of the terminology, likely less so.

Reply:
One of the quirks I'm trying to correct that the spec currently has an abstract, then an introduction, then an overview. This seems like too much setup. The middle should begin building concepts one on top of another.

In technical writing, we need to accommodate audiences from a variety of backgrounds. Terms should be introduced so they are understandable by the layman, at least so you get a feeling quickly that you're not in the target audience for the spec. e.g. If I am looking for a way to write computation in JSON, I can read the introduction and understand that's not what this does.

It should also define its relationships to other technical terminology, to accommodate the technical audiences. People familiar with finite state machines will read this and may wonder if what we are defining is different than an acceptor, [and it is not](https://en.wikipedia.org/wiki/Finite-state_machine#Acceptors). But the fact the introduction says "also known as an acceptor" should not throw off the layman.

Reply:
> One of the quirks I'm trying to correct that the spec currently has an abstract, then an introduction, then an overview. This seems like too much setup. The middle should begin building concepts one on top of another

I didn't read all of the surrounding sections again, but this would indeed sound problematic to me and like the sort of thing I'd agree we should address.

The rest of what you mentioned though, I'd stick with what I said personally. I know what a finite state machine is, I still don't find the references you're adding helpful. Again, my opinion.

Reply:
I agree with Julian.  Even if people know what an FSM is, significantly fewer people have a real understanding of them or how a JSON Schema can be mapped into one.  I think these are more theory-oriented ideas that would be better suited for supplemental literature rather than being in the spec itself.

Reply:
> I know what a finite state machine is, I still don't find the references you're adding helpful

> significantly fewer people have a real understanding of them or how a JSON Schema can be mapped into one

Ok, though my argument is that not every part of the intro has to be helpful to everyone; it has to be written so that the widest possible audience will understand what JSON Schema accomplishes for them.

The two biggest audiences, I think, will be application developers ("I want a DSL for checking JSON, instead of doing it in code") and formal grammars ("I know what ABNF and DTDs are, I want this for JSON").

I think you'll find that other similar technology uses technical terms much _more_ heavily than I'm suggesting we do.

I looked at the [introduction for ABNF](https://www.rfc-editor.org/rfc/rfc5234#section-1), which I found far too technical for most people to understand. It talks in technical terms that it's a formal syntax, but doesn't really describe why you'd want to use it at all, or use it over other languages.

[XML DTD also talks about formal grammars](https://www.w3.org/TR/xml11/#sec-prolog-dtd), validators, and uses the accepts/rejects terminology; but it too is somewhat technical and it's not immediately obvious to me who the target audience is.

So what I'm looking for is (1) is accommodating the audience familiar with formal grammars actually unimportant? (Since ABNF and DTDs both seem to be written exclusively for this audience, I would suggest no.)

And (2) if we should accommodate the formal grammar audience, is there a better way to write it so that it's more helpful for them, and less confusing to others?

Reply:
> not simply offering my own anecdote about my personal reading

Ok, I ask because saying "I don't find it helpful" is suggestive of a personal opinion without projecting what others will think; saying "I don't believe this will be helpful" is a general observation of the sort I'm looking for.

I'm going to have to think about what else to say, if it's not immediately obvious that formal grammars are related here, as that's the formal study of what JSON Schema is fundamentally doing.

>  I also don't see the "accepts/rejects" terminology in the section you linked. It uses "valid", as we already do.

XML does not use the term "validates" (in the third person singular) to refer to an outcome (and actually it doesn't use it in that form at all). It uses "validate" to describe a process, "accept"/"matches", and "reject" to describe outcomes of that process, and "valid" to describe documents that have been accepted by the process, but nothing like "validates successfully" as we do.

Reply:
> Ok, I ask because saying "I don't find it helpful" is suggestive of a personal opinion without projecting what others will think; saying "I don't believe this will be helpful" is a general observation of the sort I'm looking for.

At the risk of quoting myself, the comment I left before that was quite clear on which I was intending, please don't ignore it:

> All in all I find the first few paragraphs here to be a step back

> I don't see them as adding understanding to someone reading the spec

> what's here in this whole PR does too much